MorphicServer/ValidateEmailEndpoint.cs

```diff
@@ -0,0 +1,141 @@
+// * Administration for Independent Living & Dept. of Education under grants 
+//   H133E080022 (RERC-IT) and H133E130028/90RE5003-01-00 (UIITA-RERC)
+// * European Union's Seventh Framework Programme (FP7/2007-2013) grant 
+//   agreement nos. 289016 (Cloud4all) and 610510 (Prosperity4All)
+// * William and Flora Hewlett Foundation
+// * Ontario Ministry of Research and Innovation
+// * Canadian Foundation for Innovation
+// * Adobe Foundation
+// * Consumer Electronics Association Foundation
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using MorphicServer.Attributes;
+
+namespace MorphicServer
+{
+    [Path("/v1/verifyEmail/{oneTimeToken}")]
+    public class ValidateEmailEndpoint : Endpoint
+    {
+        /// <summary>The lookup id to use, populated from the request URL</summary>
+        [Parameter]
+        public string oneTimeToken = "";
+        
+        /// <summary>The user data populated by <code>LoadResource()</code></summary>
+        public User User = null!;
+        /// <summary>The limited-use token data populated by <code>LoadResource()</code></summary>
+        public OneTimeToken OneTimeToken = null!;
+
+        public override async Task LoadResource()
+        {
+            var hashedToken = OneTimeToken.TokenHashedWithDefault(oneTimeToken);
+            OneTimeToken = await Load<OneTimeToken>(hashedToken);
+            if (OneTimeToken == null || !OneTimeToken.IsValid())
+            {
+                throw new HttpError(HttpStatusCode.NotFound);
+            }
+            User = await Load<User>(OneTimeToken.UserId) ?? throw new HttpError(HttpStatusCode.BadRequest);
+        }
+        
+        /// <summary>Fetch the user</summary>
+        [Method]
+        public async Task Get()
+        {
+            User.EmailVerified = true;
+            await Save(User);
+            await OneTimeToken.Invalidate(Context.GetDatabase());
+            // TODO Need to respond with a nicer webpage than ""
+        }
+
+        public class ValidateEmailEndpointException : MorphicServerException
+        {
+            protected ValidateEmailEndpointException(string error) : base(error)
+            {
+            }
+
+            protected ValidateEmailEndpointException()
+            {
+            }
+        }
+        
+        public class NoServerUrlFoundException : ValidateEmailEndpointException
+        {
+            public NoServerUrlFoundException(string error) : base(error)
+            {
+            }
+
+            public NoServerUrlFoundException()
+            {
+            }
+        }
+        
+        public class NotValidPathException : ValidateEmailEndpointException
+        {
+            public NotValidPathException(string error) : base(error)
+            {
+            }
+        }
+        
+        public static string GetEmailVerificationLinkTemplate(IHeaderDictionary requestHeaders, MorphicSettings morphicSettings)
```

Comment:
Might be nice to have a more general way of getting an URL for any endpoint given just a dictionary of URL parameters.

If you're already in an Endpoint instance, maybe something like:

````
var parameters = new Dictionary<string, string>(){
    {"oneTimeToken", someToken}
};
var url = GetUrlForEndpoint(ValidateEmailEndpoint, parameters);
````

Reply:
Already have this in another branch :) Had the same thought as you after writing this.